pkg/agent/agent.go

```diff
@@ -155,6 +156,9 @@ func (k *Klocksmith) Run() error {
 		glog.Warningf("error waiting for an ok-to-reboot: %v", err)
 	}
 
+	// stop watching the update status
+	watchUpdateStatusStop <- struct{}{}
```

Comment:
https://blog.golang.org/pipelines#explicit-cancellation is the more usual way of managing done channels. See #108 which attempts to normalize the channel handling in other places as well.

"A receive operation on a closed channel can always proceed immediately, yielding the element type's zero value. This means that [we] can unblock all the receivers simply by closing the done channel. This close is effectively a broadcast signal. We extend each of our functions to accept done as a parameter and arrange for the close to happen via a defer statement, so that all return paths from main will signal the pipeline stages to exit."